### PR TITLE
Avoids buggy gcc 9.5 with cling

### PR DESCRIPTION
**Pull request by JohanMabille**
As suggested [here](https://github.com/conda-forge/cling-feedstock/pull/53#issuecomment-1282013078)

### CODE REVIEW

**Comment by hmaarrfk** [comment]
maybe we can apply the little patch to gcc.

**Comment by hmaarrfk** [comment]
Patch applied to cling feedstock https://github.com/conda-forge/cling-feedstock/blob/main/recipe/meta.yaml#L52

File: recipe/gen_patch_json.py [hunk 1]
```diff
@@ -1970,6 +1970,14 @@ def _gen_new_index_per_key(repodata, subdir, index_key):
                 ):
             _replace_pin("python >=3.7", "python >=3.8", record["depends"], record)
 
+        # add  as run_constrained for cling
+        if (
+            record_name == "cling"
+            and record['version'] >= "0.8"
+        ):
+            record.setdefault('constrains', []).extend((
+                "gxx_linux-64 !=9.5.0  # [clangdev == \"5.*\"] - sysroot_linux-64  # [linux]"
```
Comment (hmaarrfk) [inline recipe/gen_patch_json.py:1979]:
```suggestion
                "gxx_linux-64 !=9.5.0  # [clangdev == \"5.*\"] - sysroot_linux-64  # [linux]",
```

**Review by hmaarrfk — CHANGES_REQUESTED** [review]
Current patch is malformed.

**Comment by hmaarrfk** [comment]
```
win-64::cling-0.8-h97333cc_2.tar.bz2
-  "version": "0.8"
+  "version": "0.8",
+  "constrains": [
+    "gxx_linux-64 !=9.5.0  # [clangdev == \"5.*\"] - sysroot_linux-64  # [linux]"
+  ]
win-64::cling-0.8-hab3b255_0.tar.bz2
-  "version": "0.8"
+  "version": "0.8",
+  "constrains": [
+    "gxx_linux-64 !=9.5.0  # [clangdev == \"5.*\"] - sysroot_linux-64  # [linux]"
+  ]
win-64::cling-0.8-hab3b255_1.tar.bz2
-  "version": "0.8"
+  "version": "0.8",
+  "constrains": [
+    "gxx_linux-64 !=9.5.0  # [clangdev == \"5.*\"] - sysroot_linux-64  # [linux]"
+  ]

```

File: recipe/gen_patch_json.py [hunk 2]
```diff
@@ -1970,6 +1970,14 @@ def _gen_new_index_per_key(repodata, subdir, index_key):
                 ):
             _replace_pin("python >=3.7", "python >=3.8", record["depends"], record)
 
+        # add  as run_constrained for cling
+        if (
+            record_name == "cling"
+            and record['version'] >= "0.8"
+        ):
+            record.setdefault('constrains', []).extend((
+                "gxx_linux-64 !=9.5.0  # [clangdev == \"5.*\"] - sysroot_linux-64  # [linux]",
```
Comment (hmaarrfk) [inline recipe/gen_patch_json.py:1979]:
```suggestion
                "gxx_linux-64 !=9.5.0",
```

I don't think that selectors work here. Can you update your patching?

**Comment by hmaarrfk** [comment]
Sorry to ask this, but can you confirm, by going through the log and copy pasting it here, that this indeed doing what you want it to do?

**Comment by chrisburr** [comment]
Looks good to me:

```
2022-10-18T14:00:39.6779888Z + python show_diff.py
2022-10-18T14:00:40.1267990Z Downloading: https://conda.anaconda.org/conda-forge/noarch/repodata_from_packages.json.bz2
2022-10-18T14:00:42.3573657Z Downloading: https://conda.anaconda.org/conda-forge/noarch/repodata.json.bz2
2022-10-18T14:01:02.8323383Z Downloading: https://conda.anaconda.org/conda-forge/linux-64/repodata_from_packages.json.bz2
2022-10-18T14:01:03.8608030Z Downloading: https://conda.anaconda.org/conda-forge/linux-64/repodata.json.bz2
2022-10-18T14:02:14.2911087Z linux-64::cling-0.8-he0ac6c6_2.tar.bz2
2022-10-18T14:02:14.2913246Z +    "gxx_linux-64 !=9.5.0",
2022-10-18T14:02:14.2914659Z linux-64::cling-0.8-hf817b99_0.tar.bz2
2022-10-18T14:02:14.2915802Z -  "version": "0.8"
2022-10-18T14:02:14.2916771Z +  "version": "0.8",
2022-10-18T14:02:14.2917529Z +  "constrains": [
2022-10-18T14:02:14.2918531Z +    "gxx_linux-64 !=9.5.0"
2022-10-18T14:02:14.2919344Z +  ]
2022-10-18T14:02:14.2920338Z linux-64::cling-0.8-hf817b99_1.tar.bz2
2022-10-18T14:02:14.2921661Z -  "version": "0.8"
2022-10-18T14:02:14.2922626Z +  "version": "0.8",
2022-10-18T14:02:14.2923346Z +  "constrains": [
2022-10-18T14:02:14.2924323Z +    "gxx_linux-64 !=9.5.0"
2022-10-18T14:02:14.2925123Z +  ]
2022-10-18T14:02:15.3220517Z Downloading: https://conda.anaconda.org/conda-forge/linux-armv7l/repodata_from_packages.json.bz2
2022-10-18T14:02:15.4018518Z Downloading: https://conda.anaconda.org/conda-forge/linux-armv7l/repodata.json.bz2
2022-10-18T14:02:20.7310585Z Downloading: https://conda.anaconda.org/conda-forge/linux-aarch64/repodata_from_packages.json.bz2
2022-10-18T14:02:21.6156932Z Downloading: https://conda.anaconda.org/conda-forge/linux-aarch64/repodata.json.bz2
2022-10-18T14:02:41.5670071Z linux-aarch64::cling-0.8-h6293a0b_0.tar.bz2
2022-10-18T14:02:41.5672867Z -  "version": "0.8"
2022-10-18T14:02:41.5673901Z +  "version": "0.8",
2022-10-18T14:02:41.5675836Z +  "constrains": [
2022-10-18T14:02:41.5677364Z +    "gxx_linux-64 !=9.5.0"
2022-10-18T14:02:41.5678072Z +  ]
2022-10-18T14:02:41.5679109Z linux-aarch64::cling-0.8-h6293a0b_1.tar.bz2
2022-10-18T14:02:41.5680092Z -  "version": "0.8"
2022-10-18T14:02:41.5680929Z +  "version": "0.8",
2022-10-18T14:02:41.5681530Z +  "constrains": [
2022-10-18T14:02:41.5682397Z +    "gxx_linux-64 !=9.5.0"
2022-10-18T14:02:41.5683054Z +  ]
2022-10-18T14:02:41.7658109Z Downloading: https://conda.anaconda.org/conda-forge/linux-ppc64le/repodata_from_packages.json.bz2
2022-10-18T14:02:42.0386238Z Downloading: https://conda.anaconda.org/conda-forge/linux-ppc64le/repodata.json.bz2
2022-10-18T14:03:01.3325565Z Downloading: https://conda.anaconda.org/conda-forge/osx-64/repodata_from_packages.json.bz2
2022-10-18T14:03:01.9151793Z Downloading: https://conda.anaconda.org/conda-forge/osx-64/repodata.json.bz2
2022-10-18T14:04:08.3256307Z osx-64::cling-0.8-h07d71dc_2.tar.bz2
2022-10-18T14:04:08.3258193Z -  "version": "0.8"
2022-10-18T14:04:08.3258878Z +  "version": "0.8",
2022-10-18T14:04:08.3259637Z +  "constrains": [
2022-10-18T14:04:08.3260389Z +    "gxx_linux-64 !=9.5.0"
2022-10-18T14:04:08.3260980Z +  ]
2022-10-18T14:04:08.3261744Z osx-64::cling-0.8-h1d6eb1e_0.tar.bz2
2022-10-18T14:04:08.3262740Z -  "version": "0.8"
2022-10-18T14:04:08.3263675Z +  "version": "0.8",
2022-10-18T14:04:08.3264256Z +  "constrains": [
2022-10-18T14:04:08.3265019Z +    "gxx_linux-64 !=9.5.0"
2022-10-18T14:04:08.3265667Z +  ]
2022-10-18T14:04:08.3266446Z osx-64::cling-0.8-hc1cc398_1.tar.bz2
2022-10-18T14:04:08.3267486Z -  "version": "0.8"
2022-10-18T14:04:08.3268116Z +  "version": "0.8",
2022-10-18T14:04:08.3268663Z +  "constrains": [
2022-10-18T14:04:08.3269832Z +    "gxx_linux-64 !=9.5.0"
2022-10-18T14:04:08.3270446Z +  ]
2022-10-18T14:04:09.1828460Z Downloading: https://conda.anaconda.org/conda-forge/osx-arm64/repodata_from_packages.json.bz2
2022-10-18T14:04:09.3581519Z Downloading: https://conda.anaconda.org/conda-forge/osx-arm64/repodata.json.bz2
2022-10-18T14:04:17.8438340Z Downloading: https://conda.anaconda.org/conda-forge/win-32/repodata_from_packages.json.bz2
2022-10-18T14:04:18.0045841Z Downloading: https://conda.anaconda.org/conda-forge/win-32/repodata.json.bz2
2022-10-18T14:04:23.5179792Z Downloading: https://conda.anaconda.org/conda-forge/win-64/repodata_from_packages.json.bz2
2022-10-18T14:04:23.8802780Z Downloading: https://conda.anaconda.org/conda-forge/win-64/repodata.json.bz2
2022-10-18T14:05:11.6486332Z win-64::cling-0.8-h97333cc_2.tar.bz2
2022-10-18T14:05:11.6489208Z -  "version": "0.8"
2022-10-18T14:05:11.6489986Z +  "version": "0.8",
2022-10-18T14:05:11.6490613Z +  "constrains": [
2022-10-18T14:05:11.6491443Z +    "gxx_linux-64 !=9.5.0"
2022-10-18T14:05:11.6492205Z +  ]
2022-10-18T14:05:11.6493281Z win-64::cling-0.8-hab3b255_0.tar.bz2
2022-10-18T14:05:11.6494233Z -  "version": "0.8"
2022-10-18T14:05:11.6494903Z +  "version": "0.8",
2022-10-18T14:05:11.6495526Z +  "constrains": [
2022-10-18T14:05:11.6496477Z +    "gxx_linux-64 !=9.5.0"
2022-10-18T14:05:11.6497143Z +  ]
2022-10-18T14:05:11.6497951Z win-64::cling-0.8-hab3b255_1.tar.bz2
2022-10-18T14:05:11.6498852Z -  "version": "0.8"
2022-10-18T14:05:11.6499517Z +  "version": "0.8",
2022-10-18T14:05:11.6500102Z +  "constrains": [
2022-10-18T14:05:11.6500913Z +    "gxx_linux-64 !=9.5.0"
2022-10-18T14:05:11.6501545Z +  ]
2022-10-18T14:05:11.7392222Z + exit 0
```

**Pull request merged**